icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -5323,4 +5324,27 @@ void RBBITest::TestTraceCreateBreakEngine(void) {
 }
 #endif
 
+void RBBITest::TestUnpairedSurrogate() {
+    UnicodeString rules(u"ab;");
+
+    UErrorCode status = U_ZERO_ERROR;
+    UParseError pe;
+    RuleBasedBreakIterator bi1(rules, pe, status);
+    assertSuccess(WHERE, status);
+    UnicodeString rtRules = bi1.getRules();
+    // make sure the simple one work first.
+    assertEquals(WHERE, UnicodeString(u"ab;"),  rtRules);
+
+    char16_t r2[] = {'a', 0xd800, ';', 0};
```

Comment:
Would the following work -
  UnicodeString rules2("a\\\\ud800;");
  rules2.unescape();

Reply:
please use Andy's version

Reply:
Correction: unescape() is const and returns a new string, so
```
UnicodeString rules2 = UnicodeString(u"a\\ud800;").unescape();
```